modules/core/src/main/java/org/apache/ignite/internal/processors/service/ServiceClusterDeploymentResultBatch.java

```diff
@@ -0,0 +1,125 @@
+import org.apache.ignite.internal.util.tostring.GridToStringExclude;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.lang.IgniteUuid;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Batch of services cluster deployment result.
+ * <p/>
+ * Contains collection of {@link ServiceClusterDeploymentResult}.
+ */
+public class ServiceClusterDeploymentResultBatch implements DiscoveryCustomMessage {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Unique custom message ID. */
+    private final IgniteUuid id = IgniteUuid.randomUuid();
+
+    /** Deployment process id. */
+    private final ServiceDeploymentProcessId depId;
+
+    /** Services deployments results. */
+    @GridToStringInclude
+    private Collection<ServiceClusterDeploymentResult> results;
+
+    /** Services deployment actions to be processed on services deployment process. */
+    @GridToStringExclude
+    @Nullable private transient ServiceDeploymentActions serviceDeploymentActions;
+
+    /**
+     * @param depId Deployment process id.
+     * @param results Services deployments results.
+     */
+    public ServiceClusterDeploymentResultBatch(@NotNull ServiceDeploymentProcessId depId,
+        @NotNull Collection<ServiceClusterDeploymentResult> results) {
+        this.depId = depId;
+        this.results = results;
+    }
+
+    /**
+     * @return Deployment process id.
+     */
+    public ServiceDeploymentProcessId deploymentId() {
+        return depId;
+    }
+
+    /**
+     * @return Services deployments results.
+     */
+    public Collection<ServiceClusterDeploymentResult> results() {
+        return results;
+    }
+
+    /**
+     * @return Services deployment actions to be processed on services deployment process.
+     */
+    @Nullable public ServiceDeploymentActions servicesDeploymentActions() {
+        return serviceDeploymentActions;
+    }
+
+    /**
+     * @param serviceDeploymentActions Services deployment actions to be processed on services deployment process.
+     */
+    public void servicesDeploymentActions(ServiceDeploymentActions serviceDeploymentActions) {
+        this.serviceDeploymentActions = serviceDeploymentActions;
+    }
+
+    /** {@inheritDoc} */
+    @Override public IgniteUuid id() {
+        return id;
+    }
+
+    /** {@inheritDoc} */
+    @Nullable @Override public DiscoveryCustomMessage ackMessage() {
+        // No-op.
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean isMutable() {
+        return false;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean stopProcess() {
+        return false;
+    }
+
+    /** {@inheritDoc} */
+    @Override public DiscoCache createDiscoCache(GridDiscoveryManager mgr, AffinityTopologyVersion topVer,
+        DiscoCache discoCache) {
+        // No-op.
```

Comment:
No need in this comment.

Reply:
It has been added to be similar to Ignite's codebase where it is used widely.

Reply:
I thought we only use it in `void` methods (to have at least some body), but OK, I don't mind having that here either.